src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -1018,64 +1063,65 @@ protected virtual string GetOutputFileName(Compilation compilation, Cancellation
             }
             catch (Exception e)
             {
-                MessageProvider.ReportStreamWriteException(e, filePath, consoleOutput);
+                MessageProvider.ReportStreamWriteException(e, filePath, diagnostics);
                 return null;
             }
         }
 
-        protected Stream GetWin32Resources(CommandLineArguments arguments, Compilation compilation, out IEnumerable<DiagnosticInfo> errors)
+        protected Stream GetWin32Resources(CommandLineArguments arguments, Compilation compilation, DiagnosticBag diagnostics)
         {
-            return GetWin32ResourcesInternal(MessageProvider, arguments, compilation, out errors);
+            return GetWin32ResourcesInternal(MessageProvider, arguments, compilation, diagnostics);
         }
 
         // internal for testing
         internal static Stream GetWin32ResourcesInternal(CommonMessageProvider messageProvider, CommandLineArguments arguments, Compilation compilation, out IEnumerable<DiagnosticInfo> errors)
```

Comment:
Nit: I think we could change IEnumerable to ImmutableArray in `GetWin32ResourcesInternal` and `GetWin32Resources` signatures too